### DISCUSSION TITLE
Minio S3 settings

### DISCUSSION THREAD

**Discussion by gregology** [mastodon/mastodon#20386] [Q&A] [answered]
Hello Team,

I'm running a Minio instance for S3 storage but I'm unable to connect to it in Mastodon. I can connect to my Minio instance in an S3 browser with these settings

<img width="450" alt="image" src="https://user-images.githubusercontent.com/1595448/201245432-b738e4a1-fdcd-4283-b8c1-66589b0ca174.png">

However, these settings fail to connect (on 3.5.3) and I can't find any documentation on connecting to non AWS S3 buckets.

```
S3_ENABLED=true
S3_BUCKET=mastodon
AWS_ACCESS_KEY_ID=mastodon
AWS_SECRET_ACCESS_KEY=mysupersmartpassword
S3_ALIAS_HOST=s3.my.domain
```

@leahoswald, you mentioned connecting to a Minio instance in https://github.com/mastodon/mastodon/discussions/19500. Do you have any insights?

**Comment by leahoswald** [comment, marked as answer]
Just short because I'm out of time now. Maybe these help you: https://github.com/cybrespace/cybrespace-meta/blob/master/s3.md
and https://stanislas.blog/2018/05/moving-mastodon-media-files-to-wasabi-object-storage/

**Reply by gregology** [reply]
Cheers! Those links got me on the right direction. These were the settings I used with my Minio instance.

```
S3_ENABLED=true
S3_BUCKET=mastodon
AWS_ACCESS_KEY_ID=user
AWS_SECRET_ACCESS_KEY=pass
S3_PROTOCOL=https
S3_HOSTNAME=s3.my.domain
S3_ENDPOINT=https://s3.my.domain/
S3_FORCE_SINGLE_REQUEST=true
```
